WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -0,0 +1,327 @@
+
+                if (mCurrentFilter == selectedCriteria) {
+                    AppLog.d(mTAG, "The selected STATUS is already active: " +
+                            selectedCriteria.getLabel());
+                    return;
+                }
+
+                AppLog.d(mTAG, "NEW STATUS : " + selectedCriteria.getLabel());
+                setCurrentFilter(selectedCriteria);
+                if (mListener != null) {
+                    mListener.onFilterSelected(position, selectedCriteria);
+                    setRefreshing(true);
+                    mListener.onLoadData();
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+                // nop
+            }
+        });
+
+    }
+
+    private boolean hasAdapter() {
+        return (mAdapter != null);
+    }
+
+    public boolean emptyViewIsVisible(){
+        return (mEmptyView != null && mEmptyView.getVisibility() == View.VISIBLE);
+    }
+
+    public void hideEmptyView() {
+        if (mEmptyView != null) {
+            mEmptyView.setVisibility(View.GONE);
+        }
+    }
+
+    public void updateEmptyView(EmptyViewMessageType emptyViewMessageType) {
+        if (!hasAdapter() || mEmptyView == null) return;
+
+        if ((hasAdapter() && mAdapter.getItemCount() == 0) || !hasAdapter()) {
+            String msg = null;
+            if (mListener != null){
+                msg  = mListener.onShowEmptyViewMessage(emptyViewMessageType);
+            }
+
+            if (msg == null){
+                msg = getContext().getString(R.string.empty_list_default);
+            }
+            mEmptyView.setText(msg);
+            mEmptyView.setVisibility(View.VISIBLE);
+
+        } else {
+            mEmptyView.setVisibility(View.GONE);
+        }
+    }
+
+    /**
+     * show/hide progress bar which appears at the bottom when loading more items
+     */
+    public void showLoadingProgress() {
+        if (mProgressLoadMore != null) {
+            mProgressLoadMore.setVisibility(View.VISIBLE);
+        }
+    }
+
+    public void hideLoadingProgress() {
+        if (mProgressLoadMore != null) {
+            mProgressLoadMore.setVisibility(View.GONE);
+        }
+    }
+
+
+    /*
+     * adapter used by the filter spinner
+     */
+    private class SpinnerAdapter extends BaseAdapter {
+        private final FilterCriteria[] mFilterValues;
+        private final LayoutInflater mInflater;
+
+        SpinnerAdapter(Context context, FilterCriteria[] filterValues) {
+            super();
+            mInflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            mFilterValues = filterValues;
+        }
+
+        @Override
+        public int getCount() {
+            return (mFilterValues != null ? mFilterValues.length : 0);
+        }
+
+        @Override
+        public Object getItem(int position) {
+            if (position < 0 || position >= getCount())
```

Comment:
Returning an empty string when an invalid position is passed could hide some bugs. What do you think about skipping this position check so an `ArrayIndexOutOfBoundsException` occurs if an invalid position is passed?